src/Neo/SmartContract/Native/PolicyContract.cs

```diff
@@ -258,6 +260,85 @@ public bool IsBlocked(IReadOnlyStore snapshot, UInt160 account)
+
+                if (item != null)
+                {
+                    // Check UpdateCounter
+
+                    if (item.UpdateCounter == currentContract.UpdateCounter)
+                    {
+                        fixedFee = item.FixedFee;
+                        return true;
+                    }
+                }
+            }
+
+            fixedFee = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Remove whitelisted Fee contracts
+        /// </summary>
+        /// <param name="engine">The execution engine.</param>
+        /// <param name="contractHash">The contract to set the whitelist</param>
+        /// <param name="method">Method</param>
+        /// <param name="argCount">Argument count</param>
+        [ContractMethod(Hardfork.HF_Faun, CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States)]
+        private void RemoveWhitelistFeeContract(ApplicationEngine engine, UInt160 contractHash, string method, int argCount)
+        {
+            if (!CheckCommittee(engine)) throw new InvalidOperationException("Invalid committee signature");
+
+            engine.SnapshotCache.Delete(CreateStorageKey(Prefix_WhitelistedFeeContracts, contractHash, method, argCount));
+        }
+
+        /// <summary>
+        /// Set whitelisted Fee contracts
+        /// </summary>
+        /// <param name="engine">The execution engine.</param>
+        /// <param name="contractHash">The contract to set the whitelist</param>
+        /// <param name="method">Method</param>
+        /// <param name="argCount">Argument count</param>
+        /// <param name="fixedFee">Fixed execution fee</param>
+        [ContractMethod(Hardfork.HF_Faun, CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States)]
+        internal void SetWhitelistFeeContract(ApplicationEngine engine, UInt160 contractHash, string method, int argCount, long fixedFee)
+        {
+            ArgumentOutOfRangeException.ThrowIfNegative(fixedFee, nameof(fixedFee));
+
+            if (!CheckCommittee(engine)) throw new InvalidOperationException("Invalid committee signature");
+
+            var key = CreateStorageKey(Prefix_WhitelistedFeeContracts, contractHash, method, argCount);
+
+            // Validate methods
+            var contract = ContractManagement.GetContract(engine.SnapshotCache, contractHash)
+                    ?? throw new InvalidOperationException("Is not a valid contract");
+
+            if (contract.Manifest.Abi.GetMethod(method, argCount) == null)
+                throw new InvalidOperationException($"{method} with {argCount} args is not a valid method of {contractHash}");
+
+            // Set
+            var entry = engine.SnapshotCache
+                    .GetAndChange(key, () => new StorageItem(new WhitelistedFeeContract()))
+                    .GetInteroperable<WhitelistedFeeContract>();
+
+            entry.UpdateCounter = contract.UpdateCounter;
+            entry.FixedFee = fixedFee;
```

Comment:
We need add an event at the end to make Dapps know this action.
Applicationlog for this tx now:
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "txid": "0x4f76cb5a57c9f302930f50f78d1494cf5b9db780e4aea90ecd095a2226a94917",
        "executions": [
            {
                "trigger": "Application",
                "vmstate": "HALT",
                "exception": null,
                "gasconsumed": "67621",
                "stack": [
                    {
                        "type": "Any"
                    }
                ],
                "notifications": []
            }
        ]
    }
}
```

Reply:
done